src/services/narrowing.ts

```diff
@@ -0,0 +1,83 @@
+import { SelectedTableRow } from '../Components/Table/LogLineCellComponent';
```

Comment:
This is great! I think it would be important to unit test these new functions, because they are doing a lot of stuff, and would also document how we expect to use them.

Reply:
Yeah agreed, I can add tests for the 6 exported functions in here when I get back from the holiday break.

Reply:
I wouldn't mind doing it as well, btw.

Reply:
It's all yours if you want it! :)